lib/s3middleware/tagging.ts

```diff
@@ -106,6 +117,54 @@ function _validateTags(tags: Array<{ Key: string[], Value: string[] }>) {
     return tagsResult;
 }
 
+/** validateTags - Validate tags, throwing an error if tags are invalid
+* @param tags - tags parsed from xml to be validated
+* @param tags[].Key - Name of the tag
+* @param tags[].Value - Value of the tag
+* @return tagsResult - return object tags on success
+* { key: value }; error on failure
+*/
+export function validateTags(tags: Array<BucketTag>) {
+    if (tags.length === 0) {
+        return {};
+    }
+    // Maximum number of tags per resource: 50
+    if (tags.length > 50) {
+        return errorBadRequestLimit50();
+    }
+
+    const tagsResult = {};
+    for (const tag of tags) {
+        if (!_validator.validateTagObjectStructure(tag)) {
+            throw errors.MalformedXML;
+        }
+        const key = tag.Key;
+        const value = tag.Value;
+
+        if (!key) {
+            throw errors.InvalidTag.customizeDescription('The TagKey you ' +
+            'have provided is invalid');
+        }
+
+        // Allowed characters are letters, whitespace, and numbers, plus
+        // the following special characters: + - = . _ : /
+        // Maximum key length: 128 Unicode characters
+        // Maximum value length: 256 Unicode characters
```

Comment:
This description more belongs above the validate key value function then it does here. That is where we would look if we wanted to know what is being validated, not in one of the uses.

Reply:
Yes! https://github.com/scality/Arsenal/pull/1933/commits/4b19cc2101f4a15666f8746ad8fb12533e9b45d8#diff-7aa7eeaed88643ea97d57a5f15ddc2e64c380190641c53bb0c3f64d67807a475R40